contracts/Aavegotchi/facets/ItemsFacet.sol

```diff
@@ -182,41 +185,78 @@ contract ItemsFacet is Modifiers {
     ///@dev A wearable cannot be equipped in the wrong slot
     ///@param _tokenId The identifier of the aavegotchi to make changes to
     ///@param _wearablesToEquip An array containing the identifiers of the wearables to equip
-    function equipWearables(uint256 _tokenId, uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip)
-        external
-        onlyAavegotchiOwner(_tokenId)
-        onlyUnlocked(_tokenId)
-    {
+    function equipWearables(uint256 _tokenId, uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip) external {
+        uint256[EQUIPPED_WEARABLE_SLOTS] memory _depositIds;
+        _equipWearables(_tokenId, _wearablesToEquip, _depositIds);
+    }
+
+    ///@notice Allow the owner of a claimed aavegotchi to equip/unequip wearables to his aavegotchi
+    ///@dev Only valid for claimed aavegotchis
+    ///@dev A zero value will unequip that slot and a non-zero value will equip that slot with the wearable whose identifier is provided
+    ///@dev A wearable cannot be equipped in the wrong slot
+    ///@param _tokenId The identifier of the aavegotchi to make changes to
+    ///@param _wearablesToEquip An array containing the identifiers of the wearables to equip
+    ///@param _depositIds An array containing the identifiers of the deposited wearables to equip
+    function equipDelegatedWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip,
+        uint256[EQUIPPED_WEARABLE_SLOTS] calldata _depositIds
+    ) external {
+        _equipWearables(_tokenId, _wearablesToEquip, _depositIds);
+    }
+
+    function _equipWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip,
+        uint256[EQUIPPED_WEARABLE_SLOTS] memory _depositIdsToEquip
+    ) internal onlyAavegotchiOwner(_tokenId) onlyUnlocked(_tokenId) {
         Aavegotchi storage aavegotchi = s.aavegotchis[_tokenId];
+
+        // Get the GotchiEquippedDepositsInfo struct
+        GotchiEquippedDepositsInfo storage gotchiDepositInfo = s.gotchiEquippedDepositsInfo[_tokenId];
+
+        // Only valid for claimed aavegotchis
         require(aavegotchi.status == LibAavegotchi.STATUS_AAVEGOTCHI, "LibAavegotchi: Only valid for AG");
+
+        //todo: move this event to the bottom
         emit EquipWearables(_tokenId, aavegotchi.equippedWearables, _wearablesToEquip);
+        emit EquipDelegatedWearables(_tokenId, gotchiDepositInfo.equippedDepositIds, _depositIdsToEquip);
 
```

Comment:
Do you see any issue with moving these two events to the bottom of the function? They both seem a bit strangely placed @karacurt

Reply:
Fixed [here](https://github.com/aavegotchi/aavegotchi-contracts/pull/249)